File: js2cs/js2cs/Wrapper.cs
using System;
using System.Text;
using System.Threading;

namespace JS2CS
{
    public class Wrapper
    {
        private static WebSocket webSocket;
        public static Global Controller = new Global();
        public class Global
        {
            public Sockets Socket = new Sockets();
            public int StatusCode;
            public int Index;
            public bool IsNumerical;
            public bool IsPrepared;
            public bool IsAlive;
            public class WebSocketConfig
            {
                public WebSocketConfig(string address = "127.0.0.1", int port = 8080, bool secure = false)
                {
                    this.Address = address;
                    this.Port = port;
                    this.Secure = secure;

                }
                public string Address = "ws://localhost";
                public int Port = 8080;
                public bool Secure = false;
            }
            public class Message
            {
                public Message(string input)
                {
                    Input = input;
                    Output = Encoding.UTF8.GetBytes(Input);
                }
                public string Input;
                public byte[] Output;
            }
            public class Sockets
            {
                public WebSocketConfig Configuration = new WebSocketConfig();
                public Message[] Messages;
                public Message Message;
            }
            public string GetSocketString()
            {
                string secure = !Controller.Socket.Configuration.Secure ? "ws://" : "wss://";
                return secure + Controller.Socket.Configuration.Address + ":" + Controller.Socket.Configuration.Port.ToString();
            }
        }
        public static void Start()
        {
            webSocket = CreateWebsocket();
            webSocket.Connect();
            webSocket.OnMessage += (sender, e) => {
                string[] Sanitize = SanitizeText(e.Data);
                Controller.IsNumerical = false;
                if (int.TryParse(Sanitize[0], out var parsedNumber))
                {
                    Controller.IsNumerical = true;
                    Controller.StatusCode = parsedNumber;
                    Global.Message message = CreateMessage(Sanitize[1]);
                    Controller.Socket.Message = message;
                }
                if (!Controller.IsNumerical) return;
                switch (Controller.StatusCode)
                {
                    case 202: // Accepted by Server
                        Controller.StatusCode = 200;
                        Controller.IsPrepared = true;
                        Controller.Socket.Message.Input = Controller.StatusCode + "|||OK";
                        webSocket.Send(Controller.Socket.Message.Input);
                        Console.WriteLine("202: Accepted");
                        break;
                    case 200: // Recurring
                        Controller.StatusCode = 200;
                        if (Controller.IsPrepared) { Console.WriteLine("200: OK"); Controller.IsPrepared = false; }
                        webSocket.Send(Controller.StatusCode + "|||OK");
                        break;
                }           
            };
            webSocket.OnClose += (sender, e) => Console.WriteLine("499: Client Closed Connection");
            if (!webSocket.IsAlive) Console.WriteLine("503: Service Unavailable");
            else
            {
                Console.WriteLine("201: Created");
                Global.Message message = CreateMessage("201|||Created");
                webSocket.Send(message.Input);
            }
        }
        public static Global.Message CreateMessage(string Msg) { return new Global.Message(Msg); }
        private static WebSocket CreateWebsocket() { return new WebSocket(Controller.GetSocketString()); }
        static string[] SanitizeText(string input) { return input.Split(new string[] { "|||" }, System.StringSplitOptions.None); }       
    }
}

File: JSCS_Example/Node/wrapper.js
const WebSocket = require("ws");
var Global = CreateConfiguration();
var WebSocketServer = null;

function Connect() {
    function IsNumerical(str) {
        return Boolean(str.match(/\d/));
    }
    function Sleep(ms) {
        return new Promise((resolve) => setTimeout(resolve, ms));
    }
    function SanitizeText(data) {
        return Buffer.from(data).toString().split("|||");
    }
    return new Promise(function (resolve, reject) {
        WebSocketServer = new WebSocket.Server({ port: Global.Port }, () => {
            console.log("201: Created");
        });
        WebSocketServer.on("connection", async function connection(ws) {
            ws.on("close", function () {
                Global.IsAlive = false;
                console.log("499: Client Closed Connection");
            });
            ws.on("message", async function (data) {
                await Sleep(1000);
                Global.Message.Text = SanitizeText(data)[1];
                Global.Message.StatusCode = SanitizeText(data)[0];
                Global.IsNumerical = IsNumerical(Global.Message.StatusCode);
                if (!Global.IsNumerical) return;
                Global.Message.StatusCode = parseInt(SanitizeText(data)[0]);
                switch (Global.Message.StatusCode) {
                    case 201:
                        console.log("202: Accepted");
                        ws.send("202|||Accepted");
                        resolve(true);
                        break;
                    default:
                        if (Global.IsAlive) Global = CreateConfiguration();
                        else {
                            if (!Global.IsInitial) {
                                console.log("200: OK");                                
                                Global.IsInitial = true;
                            }
                            Global.IsAlive = true;                           
                        }
                        ws.send("200|||OK");
                        break;
                }
            });
        });
       
        WebSocketServer.on("error", async function connection(ws) {
            reject(new Error(err.message));
        });
    });
}

function CreateConfiguration() {
    var Global = {
        Port: 8080,
        IsNumerical: false,
        IsInitial: false,
        Message: {
            Test: "",
            StatusCode: 0,
        },
        IsAlive: false,
    };
    return Global;
}

module.exports = async () => {
    return {
        Connect: Connect,
    };
};